yt/yt/server/master/chunk_server/chunk_manager.cpp

```diff
@@ -4231,7 +4231,7 @@ class TChunkManager
                 durabilityRequired = it->second;
             } else {
                 auto replication = requisitionRegistry->GetReplication(requisitionIndex);
-                durabilityRequired = replication.IsDurabilityRequired(this);
+                durabilityRequired = replication.IsDurable(this, /*isErasureChunk*/ false);
```

Comment:
Why erasure is impossible here?

Reply:
Because of `!chunk->IsErasure()` in the if-clause on line 4254. Didn't see a good way to enforce this, also it's just a helper, so there isn't really a need.

I will add a comment here.

Reply:
Maybe pass chunk to this helper and check erasureness explicitly? It will be more robust than current implementation since someone may change usages of this helper.

Reply:
Passed the chunk to the helper and changed the behaviour here: now we can set historically_non_vital = %true for erasure chunks in some weird case (i.e. for chunks on transient mediums).